Realm/RLMQueryUtil.mm

```diff
@@ -650,14 +650,15 @@ void update_query_with_predicate(NSPredicate *predicate, RLMSchema *schema,
     else {
         // invalid predicate type
         @throw RLMPredicateException(@"Invalid predicate",
-                                     @"Only support compound, comparison and constant predicates");
+                                     @"Only support compound, comparison, and constant predicates");
     }
 }
 
 RLMProperty *RLMValidatedPropertyForSort(RLMObjectSchema *schema, NSString *propName) {
     // validate
+    RLMPrecondition(![propName containsString:@"."], @"Invalid sort property", @"Sorting on key paths is not supported.");
```

Comment:
I can't see how to fit it into a sentence in a way that makes it clear that the specific key path that was passed isn't the issue. Any suggestions?